common/src/test/kotlin/org/amshove/kluent/collections/ShouldContainSameShould.kt

```diff
@@ -0,0 +1,173 @@
+        val array = arrayOf("Hello", "World", "Wide", "Web")
+        assertFails { array shouldContainSame arrayOf("Hello", "World", "Wide", "Web", "Hello") }
+        assertFails { array shouldContainSame listOf("Hello", "World", "Wide", "Web", "Hello") }
+    }
+
+    @Test
+    fun passWhenTestingAnIntArrayWithSameValues() {
+        val array = intArrayOf(4, 3, 2, 1)
+        array shouldContainSame intArrayOf(1, 2, 3, 4)
+        array shouldContainSame listOf(1, 2, 3, 4)
+    }
+
+    @Test
+    fun failWhenTestingAnIntArrayWithDoesNotContainSameValues() {
+        val array = intArrayOf(2, 3, 4)
+        assertFails { array shouldContainSame intArrayOf(2, 3, 4, 4) }
+        assertFails { array shouldContainSame listOf(2, 3, 4, 4) }
+    }
+
+    @Test
+    fun passWhenTestingABooleanArrayWhichContainsSameValues() {
+        val array = booleanArrayOf(true, false)
+        array shouldContainSame booleanArrayOf(false, true)
+        array shouldContainSame listOf(false, true)
+    }
+
+    @Test
+    fun failWhenTestingABooleanArrayWhichDoesNotContainSameValues() {
+        val array = booleanArrayOf(false, true)
+        assertFails { array shouldContainSame booleanArrayOf(true, false, true) }
+        assertFails { array shouldContainSame listOf(true, false, true) }
+    }
+
+    @Test
+    fun passWhenTestingAByteArrayWhichContainsSameValues() {
+        val array = byteArrayOf(4, 3, 2, 1)
+        array shouldContainSame byteArrayOf(1, 2, 3, 4)
+        array shouldContainSame listOf<Byte>(1, 2, 3, 4)
+    }
+
+    @Test
+    fun failWhenTestingAByteArrayWhichDoesNotContainSameValues() {
+        val array = byteArrayOf(2, 3, 4)
+        assertFails { array shouldContainSame byteArrayOf(2, 3, 4, 4) }
+        assertFails { array shouldContainSame listOf<Byte>(2, 3, 4, 4) }
+    }
+
+    @Test
+    fun passWhenTestingACharArrayWhichContainsSameValues() {
+        val array = charArrayOf('a', 'f', 'z')
+        array shouldContainSame charArrayOf('z', 'f', 'a')
+        array shouldContainSame listOf('z', 'f', 'a')
+    }
+
+    @Test
+    fun failWhenTestingACharArrayWhichDoesNotContainSameValues() {
+        val array = charArrayOf('a', '-')
+        assertFails { array shouldContainSame charArrayOf('a', '-', 'a') }
+        assertFails { array shouldContainSame listOf('a', '-', 'a') }
+    }
+
+    @Test
+    fun passWhenTestingADoubleArrayWhichContainsSameValues() {
+        val array = doubleArrayOf(0.3, 1.1, 5.7)
+        array shouldContainSame doubleArrayOf(5.7, 0.3, 1.1)
+        array shouldContainSame listOf(5.7, 0.3, 1.1)
+    }
+
+    @Test
+    fun failWhenTestingADoubleArrayWhichDoesNotContainSameValues() {
+        val array = doubleArrayOf(0.3, 1.1, 5.7)
+        assertFails { array shouldContainSame doubleArrayOf(0.3, 1.1, 5.7, 0.3) }
+        assertFails { array shouldContainSame listOf(0.3, 1.1, 5.7, 0.3) }
+    }
+
+    @Test
+    fun passWhenTestingAFloatArrayWhichContainsSameValues() {
```

Comment:
Can you change this to
```kt
    @Test
    fun passWhenTestingAFloatArrayWhichContainsSameValues() {
        val array = floatArrayOf(5.0f, 7.0f, 3.0f)
        array shouldContainSame floatArrayOf(3.0f, 5.0f, 7.0f)
        array shouldContainSame listOf(3.0f, 5.0f, 7.0f)
    }
```

The problem is that the JavaScript runtime does have problems with equality on floating points (that's why the build is failing)

Chaning everything to `.0` will work in that case